merino/curated_recommendations/provider.py

```diff
@@ -241,10 +272,14 @@ async def fetch(
             for rank, item in enumerate(corpus_items)
         ]
 
-        # Perform publisher spread on the recommendation set
-        recommendations = CuratedRecommendationsProvider.spread_publishers(
-            recommendations, spread_distance=3
-        )
+        # 2. Perform publisher spread on the recommendation set
+        recommendations = self.spread_publishers(recommendations, spread_distance=3)
+
+        # 1. Finally, perform preferred topics boosting if preferred topics are passed in the request
+        if curated_recommendations_request.topics:
+            recommendations = self.boost_preferred_topic(
+                recommendations, curated_recommendations_request.topics
```

Comment:
A preferred topic should only be boosted up if the top 2 don't already contain a preferred topic.
```suggestion
            if not any(r.topic in curated_recommendations_request.topics for r in curated_recommendations_request[:2]):
                recommendations = self.boost_preferred_topic(
                recommendations, curated_recommendations_request.topics
```
^ Just meant to convey the basic idea. You can extract that logic into a method to improve readability. You may also want to extract 2 into a variable to avoid magic numbers.